File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0,r=height.size()-1;
        int area=0,maxArea=0;
        while(l<r){
            area=(r-l)*min(height[l],height[r]);
            maxArea=max(maxArea,area);
            if(height[l]<=height[r]){
                l++;
            } else{
                r--;
            }
        }
    return maxArea;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length, l = 0, r = n-1, area = 0;
        while(l<r){
            area = Math.max(area, (Math.min(height[l], height[r]) * (r-l)));
            if(height[l] < height[r]) l++;
            else r--;
        }
        return area;
    }
}